### PR TITLE
Eagerly load preview images (N+1)

### LINKED ISSUE #50560: No way to eager load ActiveStorage preview representations alongside variants
I was seeing N+1's when loading a page full of thumbnail preview images for a model with a `has_one_attached`, but was surprised to find #47370 already fixed it. 

Given a model like this:

```ruby
class Visual < ApplicationRecord
  has_one_attached :file, dependent: :purge_later do |attachable|
    attachable.variant :preview, resize_to_fill: [200, 200], preprocessed: true
  end
```

And querying with `Visual.with_attached_file`, looping like this:

```ruby
visual.representation(:preview)
```

If N visuals' `file` attachments are images and M are videos (i.e. generated as a [preview](https://api.rubyonrails.org/classes/ActiveStorage/Preview.html) as opposed to as a variant), then to query all of them represents an M+1. The images all eager load, but the videos don't.

I'm a total noob to ActiveStorage, but I presume this is because the `with_attached_` scopes only load variants and generated preview representations aren't quite exactly variants on the attachment. At this point the preview vs. variant separation seems to be largely vestigial as opposed to meaningful to end-users? It doesn't seem like this should be an issue.

(I ended up self-soothing for now by lazily saving each variant's blob key and am now generating a CDN-fronted URL manually to avoid the M+1.)

### PR DESCRIPTION
**Pull request by tenderlove**
For non-image attachments (like videos), generated representations are
created as a preview_image_attachment instead of as normal variants, and
as a result aren't included in the `with_attached_` scopes. This adds
those preview image attachments to the `includes` of these scopes to
avoid an N+1 when iterating over a collection of attachments and
fetching the key of their representation (variants).

Fixes #50560

### CODE REVIEW

File: activestorage/test/models/variant_with_record_test.rb [hunk 1]
```diff
@@ -105,11 +105,12 @@ class ActiveStorage::VariantWithRecordTest < ActiveSupport::TestCase
     users.reset
 
     assert_no_difference -> { ActiveStorage::VariantRecord.count } do
-      assert_queries_count(5) do
+      assert_queries_count(6) do
         # 5 queries:
```
Comment (p8) [inline activestorage/test/models/variant_with_record_test.rb:109]:
```suggestion
        # 6 queries:
```

**Comment by p8** [comment]
There is a similar scope that should probably be changed as well (maybe in another PR):
https://github.com/rails/rails/blob/af2bbd5f66ce03697dd80a1448c2c3179d7c0fda/activestorage/app/models/active_storage/attachment.rb#L45

**Comment by tenderlove** [comment]
> There is a similar scope that should probably be changed as well (maybe in another PR):
> 
> https://github.com/rails/rails/blob/af2bbd5f66ce03697dd80a1448c2c3179d7c0fda/activestorage/app/models/active_storage/attachment.rb#L45

Good call. I added a test / fix for this in this PR since I think we should backport this.

**Comment by searls** [comment]
Thanks so much for helping with this!! 🙏

**Comment by boof** [comment]
This should be part of a bigger release IMHO as this breaks apps using non-UUID PKs in their blob tables whilst the FK actually uses UUIDs w/o a deprecation warning.
A deprecation warning would be nice, since this requires a bigger migration of data.

Another option could be a documented way to disable the preloading.

**Comment by p8** [comment]
@boof Could you create an issue?

**Pull request merged**